src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -931,6 +943,15 @@ public String toString() {
     private static final ConcurrentMap<String, Addresses> cache =
         new ConcurrentHashMap<>();
 
+    /*[IF CRIU_SUPPORT]*/
+    /**
+     * To be invoked by CRIU post-restore hook, clear the cache.
+     */
+    static void clearInetAddressCache() {
+        cache.clear();
```

Comment:
should we also empty the `expirySet` ?

Reply:
`expirySet` is for removal of `cache` entries, yeah, it should be cleared as well.